horovod/tensorflow/mpi_ops.py

```diff
@@ -42,6 +46,11 @@ def _load_library(name):
     library = load_library.load_op_library(filename)
     return library
 
+try:
+    util.check_framework_version("tensorflow", tf.__version__)
+except HorovodVersionMismatch as e:
+    print("tensorflow version mismatches: \n Expected: {}. Found: {}"\
+        .format(e.installed_version, tf.__version__), file=sys.stderr)
```

Comment:
Here's my suggestion for how we sequence these things:

```
frameworks_equal, installed_version = check_installed_version('tensorflow', tf.__version__)
try:
    MPI_LIB = _load_library('mpi_lib' + get_ext_suffix())
except Exception as e:
    check_installed_version('tensorflow', tf.__version__, e)
    raise e
else:
    check_installed_version('tensorflow', tf.__version__)
```

So basically we have 4 cases:

1. Exception was raised and version mismatch -> raise a HorovodVersionMismatchError
2. Exception was raised but versions equal -> re-raise original exception
3. No exception raised and version mismatch -> issue HorovodVersionMismatchWarning
4. No exception raised but versions equal -> do nothing